skorecard/bucket_mapping.py

```diff
@@ -218,7 +220,8 @@ def transform(self, x):
         buckets = np.where(x.isnull(), self._missing_bucket, buckets)
 
         # Ensure dtype is integer buckets
-        buckets = buckets.astype(int)
+        if not np.isnan(self._missing_bucket):
```

Comment:
The `.astype(int)` is there for a reason, we had some weird bugs/edge cases and this ensures the buckets list always contain ints. Perhaps something like this could work?

```python
def to_int(x):
    if not np.isnan(x):
        return int(x)
    else:
        return x

buckets = [to_int(x) for x in buckets]
```

Reply:
added